tree/dataframe/inc/ROOT/RDFActionHelpers.hxx

```diff
@@ -246,49 +245,50 @@ extern template void
 FillHelper::Exec(unsigned int, const std::vector<unsigned int> &, const std::vector<unsigned int> &);
 
 template <typename HIST = Hist_t>
-class FillTOHelper : public RActionImpl<FillTOHelper<HIST>> {
-   std::unique_ptr<TThreadedObject<HIST>> fTo;
+class FillParHelper : public RActionImpl<FillParHelper<HIST>> {
+   std::vector<HIST*> fObjects;
 
 public:
-   FillTOHelper(FillTOHelper &&) = default;
-   FillTOHelper(const FillTOHelper &) = delete;
+   FillParHelper(FillParHelper &&) = default;
+   FillParHelper(const FillParHelper &) = delete;
 
-   FillTOHelper(const std::shared_ptr<HIST> &h, const unsigned int nSlots)
-      : fTo(std::make_unique<TThreadedObject<HIST>>(*h))
+   FillParHelper(const std::shared_ptr<HIST> &h, const unsigned int nSlots)
+      : fObjects(nSlots, nullptr)
    {
-      fTo->SetAtSlot(0, h);
+      fObjects[0] = h.get();
```

Comment:
if `FillParHelper` does not retain ownership of the contents of `h`, it should not take it by `shared_ptr` in its ctor, it's misleading.

it's not _wrong_, because the helper will go out of scope if the corresponding `RResultPtr` goes out of scope, and the `RResultPtr` shares ownership of the contents of `h` -- so I don't think it can ever happen that `FillParHelper` fills a deleted histogram -- but ownership would be clearer if `FillTOHelper`'s ctor took the histogram by reference (and then stored it in a vector of raw pointers). Is that doable?

Reply:
Perhaps. Not a good reason not to change this but this signature resisted all refactorings since the first implementation 2y ago.